bunsen/bunsen-uscore-resources/src/main/java/com/cerner/bunsen/UsCoreProfileProvider.java

```diff
@@ -0,0 +1,183 @@
+      FhirContext context) {
+    Preconditions.checkArgument(context.getVersion().getVersion() == FhirVersionEnum.DSTU3
+        || context.getVersion().getVersion() == FhirVersionEnum.R4);
+
+    IParser parser = context.newJsonParser();
+
+    if (context.getVersion().getVersion() == FhirVersionEnum.DSTU3) {
+      // TODO remove non-StructureDefinition files in definitions-stu3/ if they are not needed.
+      load(support, parser, "definitions-stu3/StructureDefinition-us-core-allergyintolerance.json");
+      load(support, parser, "definitions-stu3/StructureDefinition-us-core-birthsex.json");
+      load(support, parser, "definitions-stu3/StructureDefinition-us-core-careplan.json");
+      load(support, parser, "definitions-stu3/StructureDefinition-us-core-careteam.json");
+      load(support, parser, "definitions-stu3/StructureDefinition-us-core-condition.json");
+      load(support, parser, "definitions-stu3/StructureDefinition-us-core-device.json");
+      load(support, parser, "definitions-stu3/StructureDefinition-us-core-diagnosticreport.json");
+      load(support, parser, "definitions-stu3/StructureDefinition-us-core-direct.json");
+      load(support, parser, "definitions-stu3/StructureDefinition-us-core-documentreference.json");
+      load(support, parser, "definitions-stu3/StructureDefinition-us-core-encounter.json");
+      load(support, parser, "definitions-stu3/StructureDefinition-us-core-ethnicity.json");
+      load(support, parser, "definitions-stu3/StructureDefinition-us-core-goal.json");
+      load(support, parser, "definitions-stu3/StructureDefinition-us-core-immunization.json");
+      load(support, parser, "definitions-stu3/StructureDefinition-us-core-location.json");
+      load(support, parser, "definitions-stu3/StructureDefinition-us-core-medication.json");
+      load(support, parser, "definitions-stu3/StructureDefinition-us-core-medicationrequest.json");
+      load(
+          support, parser, "definitions-stu3/StructureDefinition-us-core-medicationstatement.json");
+      load(support, parser, "definitions-stu3/StructureDefinition-us-core-observationresults.json");
+      load(support, parser, "definitions-stu3/StructureDefinition-us-core-organization.json");
+      load(support, parser, "definitions-stu3/StructureDefinition-us-core-patient.json");
+      load(support, parser, "definitions-stu3/StructureDefinition-us-core-practitioner.json");
+      load(support, parser, "definitions-stu3/StructureDefinition-us-core-practitionerrole.json");
+      load(support, parser, "definitions-stu3/StructureDefinition-us-core-procedure.json");
+      load(support, parser, "definitions-stu3/StructureDefinition-us-core-profile-link.json");
+      load(support, parser, "definitions-stu3/StructureDefinition-us-core-race.json");
+      load(support, parser, "definitions-stu3/StructureDefinition-us-core-smokingstatus.json");
+    }
+
+    if (context.getVersion().getVersion() == FhirVersionEnum.R4) {
```

Comment:
for better readability, probably we can break this function into two one for loading DSTU3 and the other for R4